zerver/openapi/zulip.yaml

```diff
@@ -25728,6 +25740,20 @@ components:
 
             **Changes**: New in Zulip 10.0 (feature level 296).
           default: true
+        use_for_user_matching:
+          type: boolean
+          description: |
+            Whether this custom profile field should be used to match users in @ mention
+            suggestions.
+
+            This setting is only applicable to profile field types that store text values:
+            **Short text (type 1)** and **External account (type 7)**.
```

Comment:
I don't think it is needed to mention 1 and 7 values in the type. And maybe we can format this similar to how we format description for `display_in_profile_summary`.